tests/components/meteo_france/test_config_flow.py

```diff
@@ -0,0 +1,151 @@
+"""Tests for the Meteo-France config flow."""
+from unittest.mock import patch
+
+from meteofrance.client import meteofranceError
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.components.meteo_france.const import CONF_CITY, DOMAIN, SENSOR_TYPES
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.const import CONF_MONITORED_CONDITIONS
+
+from tests.common import MockConfigEntry
+
+CITY_1_POSTAL = "74220"
+CITY_1_NAME = "La Clusaz"
+CITY_2_POSTAL_DISTRICT_1 = "69001"
+CITY_2_POSTAL_DISTRICT_4 = "69004"
+CITY_2_NAME = "Lyon"
+MONITORED_CONDITIONS = ["temperature", "weather"]
+DEFAULT_MONITORED_CONDITIONS = list(SENSOR_TYPES)
+
+
+@pytest.fixture(name="client_1")
+def mock_controller_client_1():
+    """Mock a successful client."""
+    with patch(
+        "homeassistant.components.meteo_france.config_flow.meteofranceClient",
+        update=False,
+    ) as service_mock:
+        service_mock.return_value.get_data.return_value = {"name": CITY_1_NAME}
+        yield service_mock
+
+
+@pytest.fixture(name="client_2")
+def mock_controller_client_2():
+    """Mock a successful client."""
+    with patch(
+        "homeassistant.components.meteo_france.config_flow.meteofranceClient",
+        update=False,
+    ) as service_mock:
+        service_mock.return_value.get_data.return_value = {"name": CITY_2_NAME}
+        yield service_mock
```

Comment:
@MartinHjelmare Is it possible to make conditional tests with MagicMock ?

That would be like : 
```python
if service_mock.return_value.postal_code == CITY_1_POSTAL:
    service_mock.return_value.get_data.return_value = {"name": CITY_1_NAME}
else if service_mock.return_value.postal_code == CITY_2_POSTAL:
    service_mock.return_value.get_data.return_value = {"name": CITY_2_NAME}
```

Looked but can't find.

Reply:
Depends on what the attributes on the mock return. By default they return mocks, so they need to be set or specced by something.